javascript/ql/test/query-tests/Security/CWE-079/XssWithAdditionalSources.ql

```diff
@@ -0,0 +1,24 @@
+/**
+ * @name Client-side cross-site scripting
+ * @description Writing user input directly to the DOM allows for
+ *              a cross-site scripting vulnerability.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id js/xss
+ * @tags security
+ *       external/cwe/cwe-079
+ *       external/cwe/cwe-116
+ */
+
+import javascript
+import semmle.javascript.security.dataflow.DomBasedXss::DomBasedXss
+import DataFlow::PathGraph
+import semmle.javascript.heuristics.AdditionalSources
+
+
+from Configuration cfg, DataFlow::PathNode source, DataFlow::PathNode sink
+where cfg.hasFlowPath(source, sink)
```

Comment:
Can we add:

```
and source.getNode() instanceof RemoteServerResponse
``` 

or perhaps:
```
and source.getNode() instanceof HeuristicSource
```
to avoid reporting the ordinary XSS results twice in the tests?